securecookie.go

```diff
@@ -325,6 +328,7 @@ func (s *SecureCookie) Decode(name, value string, dst interface{}) error {
 	if t1, err = strconv.ParseInt(string(parts[0]), 10, 64); err != nil {
 		return errTimestampInvalid
 	}
+	s.LastTimestamp = t1
```

Comment:
What happens if another `Decode` call on a subsequent request changes this before you access the `s.LastTimestamp` field? The race condition still exists.